browsermob-core/src/main/java/net/lightbody/bmp/BrowserMobProxyServer.java

```diff
@@ -937,6 +937,16 @@ public void addResponseFilter(ResponseFilter filter) {
         addLastHttpFilterFactory(new ResponseFilterAdapter.FilterSource(filter));
     }
 
+    /**
+     * <b>Note:</b> The current implementation of this method forces a maximum response size of 2 MiB. To adjust the maximum response size, or
+     * to disable aggregation (which disallows access to the {@link net.lightbody.bmp.util.HttpMessageContents}), you may add the filter source
+     * directly: <code>addFirstHttpFilterFactory(new ResponseFilterAdapter.FilterSource(filter, bufferSizeInBytes));</code>
+     */
+    @Override
+    public void addResponseFilterWithCustomBufferSize(ResponseFilter filter, int bufferSize) {
+        addLastHttpFilterFactory(new ResponseFilterAdapter.FilterSource(filter, bufferSize));
```

Comment:
What's difference between addLast or addFirst ?

Reply:
For modifying request we use `addFirstHttpFilterFactory` and for modifying response we used `addLastHttpFilterFactory`